pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1214,6 +1216,32 @@ func (l *LibvirtDomainManager) UnfreezeVMI(vmi *v1.VirtualMachineInstance) error
 	return nil
 }
 
+func (l *LibvirtDomainManager) SoftRebootVMI(vmi *v1.VirtualMachineInstance) error {
+	domainRebootFlagValues := libvirt.DOMAIN_REBOOT_GUEST_AGENT
+	condManager := controller.NewVirtualMachineInstanceConditionManager()
+	if !condManager.HasConditionWithStatus(vmi, v1.VirtualMachineInstanceAgentConnected, k8sv1.ConditionTrue) {
+		domainRebootFlagValues = libvirt.DOMAIN_REBOOT_ACPI_POWER_BTN
```

Comment:
note, it's possible for acpi to be disabled on a VMI.

Reply:
updated